joint_state_controller/test/test_joint_state_controller.cpp

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020 PAL Robotics SL.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+
+#include "joint_state_controller/joint_state_controller.hpp"
+
+#include "lifecycle_msgs/msg/state.hpp"
+
+#include "rclcpp/utilities.hpp"
+
+#include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
+
+#include "test_joint_state_controller.hpp"
+
+#include "test_robot_hardware/test_robot_hardware.hpp"
+
+using std::placeholders::_1;
+
+namespace
+{
+constexpr rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn NODE_SUCCESS =
```

Comment:
```suggestion
constexpr auto NODE_SUCCESS =
```
does this break it all?